frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/ResetCurrentToRemoteBrackgroundable.java

```diff
@@ -0,0 +1,100 @@
+
+import com.intellij.dvcs.DvcsUtil;
+import com.intellij.openapi.application.AccessToken;
+import com.intellij.openapi.progress.ProgressIndicator;
+import com.intellij.openapi.progress.Task;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.vcs.VcsNotifier;
+import com.intellij.openapi.vfs.VfsUtil;
+import git4idea.commands.Git;
+import git4idea.commands.GitCommand;
+import git4idea.commands.GitLineHandler;
+import git4idea.commands.GitLocalChangesWouldBeOverwrittenDetector;
+import git4idea.repo.GitRepository;
+import git4idea.repo.GitRepositoryManager;
+import git4idea.util.LocalChangesWouldBeOverwrittenHelper;
+import kr.pe.kwonnam.slf4jlambda.LambdaLogger;
+import lombok.CustomLog;
+import lombok.experimental.ExtensionMethod;
+import lombok.val;
+import org.apache.commons.text.StringEscapeUtils;
+
+import com.virtuslab.gitmachete.frontend.vfsutils.GitVfsUtils;
+import com.virtuslab.qual.guieffect.UIThreadUnsafe;
+
+@ExtensionMethod({GitVfsUtils.class, StringEscapeUtils.class})
+@CustomLog
+public class ResetCurrentToRemoteBrackgroundable extends Task.Backgroundable {
+
+  public LambdaLogger log() {
+    return LOG;
+  }
+
+  private final String localBranchName;
+
+  private final String remoteTrackingBranchName;
+
+  private final GitRepository gitRepository;
+
+  public ResetCurrentToRemoteBrackgroundable(Project project, String title,
+      boolean canBeCancelled,
+      String localBranchName, String remoteTrackingBranchName, GitRepository gitRepository) {
+    super(project, title, canBeCancelled);
+    this.localBranchName = localBranchName;
+    this.remoteTrackingBranchName = remoteTrackingBranchName;
+    this.gitRepository = gitRepository;
+  }
+
+  @Override
+  @UIThreadUnsafe
+  public void run(ProgressIndicator indicator) {
+    if (myProject != null && localBranchName != null && remoteTrackingBranchName != null) {
+
+      log().debug(() -> "Resetting '${localBranchName}' to '${remoteTrackingBranchName}'");
+
+      try (AccessToken ignored = DvcsUtil.workingTreeChangeStarted(myProject,
+          getString("action.GitMachete.BaseResetToRemoteAction.task-title"))) {
+        val resetHandler = new GitLineHandler(myProject, gitRepository.getRoot(), GitCommand.RESET);
+        resetHandler.addParameters("--keep");
+        resetHandler.addParameters(remoteTrackingBranchName);
+        resetHandler.endOptions();
+
+        val localChangesDetector = new GitLocalChangesWouldBeOverwrittenDetector(
+            gitRepository.getRoot(), RESET);
+        resetHandler.addLineListener(localChangesDetector);
+
+        val result = Git.getInstance().runCommand(resetHandler);
+
+        if (result.success()) {
+          VcsNotifier.getInstance(myProject).notifySuccess( /* displayId */ null,
+              /* title */ "",
+              getString("action.GitMachete.BaseResetToRemoteAction.notification.title.reset-success.HTML")
+                  .format(localBranchName));
```

Comment:
We have a strange formatting error here, for the `.format` method. This is called the same way in numerous other places of the codebase, though🧐
Any ideas?

Reply:
probably it is about missing `GitMacheteBundle.class` in the extension methods

In particular we need: `@ExtensionMethod(GitMacheteBundle.class)`